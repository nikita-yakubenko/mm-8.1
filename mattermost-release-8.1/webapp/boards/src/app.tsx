// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React, {useEffect} from 'react'
import {DndProvider} from 'react-dnd'
import {HTML5Backend} from 'react-dnd-html5-backend'
import {TouchBackend} from 'react-dnd-touch-backend'
import {History} from 'history'

import TelemetryClient from './telemetry/telemetryClient'

import FlashMessages from './components/flashMessages'
import NewVersionBanner from './components/newVersionBanner'
import {Utils} from './utils'
import {fetchMe, getMe} from './store/users'
import {useAppDispatch, useAppSelector} from './store/hooks'
import {fetchClientConfig} from './store/clientConfig'
import FocalboardRouter from './router'

import {IUser} from './user'

type Props = {
    history?: History<unknown>
}

const App = (props: Props): JSX.Element => {
    const me = useAppSelector<IUser|null>(getMe)
    const dispatch = useAppDispatch()

    useEffect(() => {
        dispatch(fetchMe())
        dispatch(fetchClientConfig())
    }, [])

    useEffect(() => {
        if (me) {
            TelemetryClient.setUser(me)
        }
    }, [me])

    return (
        <DndProvider backend={Utils.isMobile() ? TouchBackend : HTML5Backend}>
            <FlashMessages milliseconds={2000}/>
            <div id='frame'>
                <div id='main'>
                    <NewVersionBanner/>
                    <FocalboardRouter history={props.history}/>
                </div>
            </div>
        </DndProvider>
    )
}

export default React.memo(App)
