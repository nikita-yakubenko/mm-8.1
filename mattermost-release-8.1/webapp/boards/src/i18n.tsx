// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import messages_ca from 'i18n/ca.json'
import messages_de from 'i18n/de.json'
import messages_el from 'i18n/el.json'
import messages_en from 'i18n/en.json'
import messages_enAu from 'i18n/en_AU.json'
import messages_es from 'i18n/es.json'
import messages_fa from 'i18n/fa.json'
import messages_fr from 'i18n/fr.json'
import messages_hu from 'i18n/hu.json'
import messages_id from 'i18n/id.json'
import messages_it from 'i18n/it.json'
import messages_ja from 'i18n/ja.json'
import messages_ko from 'i18n/ko.json'
import messages_nl from 'i18n/nl.json'
import messages_oc from 'i18n/oc.json'
import messages_pl from 'i18n/pl.json'
import messages_ptBr from 'i18n/pt_BR.json'
import messages_ru from 'i18n/ru.json'
import messages_sv from 'i18n/sv.json'
import messages_tr from 'i18n/tr.json'
import messages_uk from 'i18n/uk.json'
import messages_zhHans from 'i18n/zh_Hans.json'
import messages_zhHant from 'i18n/zh_Hant.json'

export function getMessages(locale: string): {[key: string]: string} {
    switch (locale) {
    // case 'bg':
    //     return messages_bg // TODO missing translation sourcefile
    case 'ca':
        return messages_ca // TODO missing option in language selector
    case 'de':
        return messages_de
    case 'el':
        return messages_el // TODO missing option in language selector
    case 'en':
    default:
        return messages_en
    case 'en-AU':
        return messages_enAu
    case 'es':
        return messages_es
    case 'fa':
        return messages_fa
    case 'fr':
        return messages_fr
    case 'hu':
        return messages_hu
    case 'id':
        return messages_id // TODO missing option in language selector
    case 'it':
        return messages_it
    case 'ja':
        return messages_ja
    case 'ko':
        return messages_ko
    case 'nl':
        return messages_nl
    case 'oc':
        return messages_oc // TODO missing option in language selector
    case 'pl':
        return messages_pl
    case 'pt-BR':
        return messages_ptBr

    // case 'ro':
    //     return messages_ro // TODO missing translation sourcefile
    case 'ru':
        return messages_ru
    case 'sv':
        return messages_sv
    case 'tr':
        return messages_tr
    case 'uk':
        return messages_uk
    case 'zh-Hans':
    case 'zh-CN':
        return messages_zhHans
    case 'zh-Hant':
    case 'zh-TW':
        return messages_zhHant
    }
}
