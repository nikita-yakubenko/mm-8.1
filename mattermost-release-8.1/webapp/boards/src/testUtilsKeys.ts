// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fireEvent} from '@testing-library/dom'

export const keyDownEscapeLegacy = (element: Element) => {
    fireEvent.keyDown(element, {
        key: 'Escape',
        code: 'Escape',
        keyCode: 27,
        charCode: 27,
    })
}

export const keyDownEnterLegacy = (element: Element) => {
    fireEvent.keyDown(element, {
        key: 'Enter',
        code: 'Enter',
        keyCode: 13,
        charCode: 13,
    })
}
