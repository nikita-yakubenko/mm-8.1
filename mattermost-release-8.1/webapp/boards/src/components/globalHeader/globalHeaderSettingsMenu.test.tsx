// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'
import {createMemoryHistory} from 'history'

import {render} from '@testing-library/react'

import userEvent from '@testing-library/user-event'
import configureStore from 'redux-mock-store'

import {mocked} from 'jest-mock'

import {wrapIntl} from 'src/testUtils'

import TelemetryClient, {TelemetryActions, TelemetryCategory} from 'src/telemetry/telemetryClient'

import client from 'src/octoClient'

import GlobalHeaderSettingsMenu from './globalHeaderSettingsMenu'

jest.mock('src/telemetry/telemetryClient')
jest.mock('src/octoClient')
const mockedTelemetry = mocked(TelemetryClient)
const mockedOctoClient = mocked(client)

describe('components/sidebar/GlobalHeaderSettingsMenu', () => {
    const mockStore = configureStore([])
    const history = createMemoryHistory()
    let store = mockStore({})
    beforeEach(() => {
        store = mockStore({
            teams: {
                current: {id: 'team_id_1'},
            },
            boards: {
                current: 'board_id',
                boards: {
                    board_id: {id: 'board_id'},
                },
                myBoardMemberships: {
                    board_id: {userId: 'user_id_1', schemeAdmin: true},
                },
            },
            users: {
                me: {
                    id: 'user-id',
                },
            },
        })
    })
    test('settings menu closed should match snapshot', () => {
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeaderSettingsMenu history={history}/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('settings menu open should match snapshot', async () => {
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeaderSettingsMenu history={history}/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        await userEvent.click(container.querySelector('.menu-entry') as Element)
        expect(container).toMatchSnapshot()
    })

    test('languages menu open should match snapshot', async () => {
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeaderSettingsMenu history={history}/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        await userEvent.click(container.querySelector('.menu-entry') as Element)
        await userEvent.hover(container.querySelector('#lang') as Element)
        expect(container).toMatchSnapshot()
    })

    test('imports menu open should match snapshot', async () => {
        window.open = jest.fn()
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeaderSettingsMenu history={history}/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        await userEvent.click(container.querySelector('.menu-entry') as Element)
        await userEvent.hover(container.querySelector('#import') as Element)
        expect(container).toMatchSnapshot()

        await userEvent.click(container.querySelector('[aria-label="Asana"]') as Element)
        expect(mockedTelemetry.trackEvent).toBeCalledWith(TelemetryCategory, TelemetryActions.ImportAsana)
    })

    test('Product Tour option restarts the tour', async () => {
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeaderSettingsMenu history={history}/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        await userEvent.click(container.querySelector('.menu-entry') as Element)
        await userEvent.click(container.querySelector('.product-tour') as Element)
        expect(mockedOctoClient.patchUserConfig).toBeCalledWith('user-id', {
            updatedFields: {
                onboardingTourStarted: '1',
                onboardingTourStep: '0',
                tourCategory: 'onboarding',
            },
        })
    })
})
