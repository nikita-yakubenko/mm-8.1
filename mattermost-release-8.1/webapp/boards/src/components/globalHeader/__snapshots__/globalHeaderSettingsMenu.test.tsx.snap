// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebar/GlobalHeaderSettingsMenu imports menu open should match snapshot 1`] = `
<div>
  <div
    class="GlobalHeaderSettingsMenu"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper override menuOpened"
      role="button"
    >
      <div
        class="GlobalHeaderComponent__button menu-entry"
      >
        <i
          class="CompassIcon icon-settings-outline SettingsIcon"
        />
      </div>
      <div
        class="Menu noselect left "
      >
        <div
          class="menu-contents"
        >
          <div
            class="menu-options"
          >
            <div>
              <div
                class="MenuOption SubMenuOption menu-option open-left menu-option-active"
                id="import"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Import
                </div>
                <i
                  class="CompassIcon icon-chevron-right"
                />
                <div
                  class="SubMenu Menu noselect left-bottom"
                >
                  <div
                    class="menu-contents"
                  >
                    <div
                      class="menu-options"
                    >
                      <div
                        aria-label="Import archive"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Import archive
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                      <div
                        aria-label="Trello"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Trello
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                      <div
                        aria-label="Asana"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Asana
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                      <div
                        aria-label="Notion"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Notion
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                      <div
                        aria-label="Jira"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Jira
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                      <div
                        aria-label="Todoist"
                        class="MenuOption TextOption menu-option"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Todoist
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                    </div>
                    <div
                      class="menu-spacer hideOnWidescreen"
                    />
                    <div
                      class="menu-options hideOnWidescreen"
                    >
                      <div
                        aria-label="Cancel"
                        class="MenuOption TextOption menu-option menu-cancel"
                        role="button"
                      >
                        <div
                          class="d-flex"
                        >
                          <div
                            class="noicon"
                          />
                        </div>
                        <div
                          class="menu-option__content"
                        >
                          <div
                            class="menu-name"
                          >
                            Cancel
                          </div>
                        </div>
                        <div
                          class="noicon"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                aria-label="Random icons"
                class="MenuOption SwitchOption menu-option"
                role="button"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Random icons
                </div>
                <div
                  class="Switch override-main size--small on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                aria-label="Product tour"
                class="MenuOption TextOption menu-option product-tour"
                role="button"
              >
                <div
                  class="d-flex"
                >
                  <div
                    class="noicon"
                  />
                </div>
                <div
                  class="menu-option__content"
                >
                  <div
                    class="menu-name"
                  >
                    Product tour
                  </div>
                </div>
                <div
                  class="noicon"
                />
              </div>
            </div>
          </div>
          <div
            class="menu-spacer hideOnWidescreen"
          />
          <div
            class="menu-options hideOnWidescreen"
          >
            <div
              aria-label="Cancel"
              class="MenuOption TextOption menu-option menu-cancel"
              role="button"
            >
              <div
                class="d-flex"
              >
                <div
                  class="noicon"
                />
              </div>
              <div
                class="menu-option__content"
              >
                <div
                  class="menu-name"
                >
                  Cancel
                </div>
              </div>
              <div
                class="noicon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/sidebar/GlobalHeaderSettingsMenu languages menu open should match snapshot 1`] = `
<div>
  <div
    class="GlobalHeaderSettingsMenu"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper override menuOpened"
      role="button"
    >
      <div
        class="GlobalHeaderComponent__button menu-entry"
      >
        <i
          class="CompassIcon icon-settings-outline SettingsIcon"
        />
      </div>
      <div
        class="Menu noselect left "
      >
        <div
          class="menu-contents"
        >
          <div
            class="menu-options"
          >
            <div>
              <div
                class="MenuOption SubMenuOption menu-option open-left"
                id="import"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Import
                </div>
                <i
                  class="CompassIcon icon-chevron-right"
                />
              </div>
            </div>
            <div>
              <div
                aria-label="Random icons"
                class="MenuOption SwitchOption menu-option"
                role="button"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Random icons
                </div>
                <div
                  class="Switch override-main size--small on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                aria-label="Product tour"
                class="MenuOption TextOption menu-option product-tour"
                role="button"
              >
                <div
                  class="d-flex"
                >
                  <div
                    class="noicon"
                  />
                </div>
                <div
                  class="menu-option__content"
                >
                  <div
                    class="menu-name"
                  >
                    Product tour
                  </div>
                </div>
                <div
                  class="noicon"
                />
              </div>
            </div>
          </div>
          <div
            class="menu-spacer hideOnWidescreen"
          />
          <div
            class="menu-options hideOnWidescreen"
          >
            <div
              aria-label="Cancel"
              class="MenuOption TextOption menu-option menu-cancel"
              role="button"
            >
              <div
                class="d-flex"
              >
                <div
                  class="noicon"
                />
              </div>
              <div
                class="menu-option__content"
              >
                <div
                  class="menu-name"
                >
                  Cancel
                </div>
              </div>
              <div
                class="noicon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/sidebar/GlobalHeaderSettingsMenu settings menu closed should match snapshot 1`] = `
<div>
  <div
    class="GlobalHeaderSettingsMenu"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper"
      role="button"
    >
      <div
        class="GlobalHeaderComponent__button menu-entry"
      >
        <i
          class="CompassIcon icon-settings-outline SettingsIcon"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/sidebar/GlobalHeaderSettingsMenu settings menu open should match snapshot 1`] = `
<div>
  <div
    class="GlobalHeaderSettingsMenu"
  >
    <div
      aria-label="menuwrapper"
      class="MenuWrapper override menuOpened"
      role="button"
    >
      <div
        class="GlobalHeaderComponent__button menu-entry"
      >
        <i
          class="CompassIcon icon-settings-outline SettingsIcon"
        />
      </div>
      <div
        class="Menu noselect left "
      >
        <div
          class="menu-contents"
        >
          <div
            class="menu-options"
          >
            <div>
              <div
                class="MenuOption SubMenuOption menu-option open-left"
                id="import"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Import
                </div>
                <i
                  class="CompassIcon icon-chevron-right"
                />
              </div>
            </div>
            <div>
              <div
                aria-label="Random icons"
                class="MenuOption SwitchOption menu-option"
                role="button"
              >
                <div
                  class="noicon"
                />
                <div
                  class="menu-name"
                >
                  Random icons
                </div>
                <div
                  class="Switch override-main size--small on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                aria-label="Product tour"
                class="MenuOption TextOption menu-option product-tour"
                role="button"
              >
                <div
                  class="d-flex"
                >
                  <div
                    class="noicon"
                  />
                </div>
                <div
                  class="menu-option__content"
                >
                  <div
                    class="menu-name"
                  >
                    Product tour
                  </div>
                </div>
                <div
                  class="noicon"
                />
              </div>
            </div>
          </div>
          <div
            class="menu-spacer hideOnWidescreen"
          />
          <div
            class="menu-options hideOnWidescreen"
          >
            <div
              aria-label="Cancel"
              class="MenuOption TextOption menu-option menu-cancel"
              role="button"
            >
              <div
                class="d-flex"
              >
                <div
                  class="noicon"
                />
              </div>
              <div
                class="menu-option__content"
              >
                <div
                  class="menu-name"
                >
                  Cancel
                </div>
              </div>
              <div
                class="noicon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
